components/integration/integration.php

```diff
@@ -418,6 +476,22 @@ public function order_status( $order_id, $status_old = 'new', $status_new = 'pen
 				if ( function_exists( $subscription_function ) ) {
 					$response = call_user_func( $subscription_function, $subscription_id, $email, $name );
 
+					if ( ! is_wp_error( $response ) && ! empty( $this->api_key ) && 'tag' === $subscription_type ) {
+                        $options = ckwc_get_subscription_options();
+                        $tags    = array();
+                        foreach ( $options as $option ) {
+	                        if ( 'tag' !== $option['key'] ) {
+		                        continue;
+	                        }
+                            $tags = $option['options'];
```

Comment:
Are you resetting the outter layer tags on each loop? Shouldn't you add to the array instead?

Reply:
I don't _think_ so. This is getting all the tags for the connected ConvertKit account, then if the `$subscription_id` being worked on (which is an ID of something in ConvertKit to subscribe someone to with a WooCommerce purchase) is in that list of tags, we add an order note that we tagged them.

But I think I see what you mean – there's no reason to initialize `$tags = array()` if we're then reassigning it to `$option['options']`. I think that _might_ be to ensure `$tags` exists at all so we can check it... but then we wouldn't need the `if ( $tags )` check there 🤔 

This is pretty messy and I don't like it now that I'm looking at it more closely. I'm going to see if I can clean it up and make it more understandable.

Reply:
So if you have 2 `$option` in `$options` that has `$option['options']` set, that means you will only set the $tags to the last $option that you went over, right?

Said differently, all previous value of `$tags` will be dumped for the latest version of it.

Reply:
Ah, I see what you mean.

The `$options` array will only ever be an array of arrays, one each with keys `course`, `form`, and `tag`, so `$tags = $option['options']` is only reached once.

But, that's not exactly clear without reading further in the source and it's sure not clean, so I'll try to include that in my cleanup efforts 😂